clients/linux/src/app.rs

```diff
@@ -296,11 +306,25 @@ impl LbApp {
     }
 
     fn open_file(&self, maybe_id: Option<Uuid>) -> LbResult<()> {
+        // Ask the user how the want to deal with unsaved changes
+        if self.state.borrow().open_file_dirty {
+            if let Some(open_file) = self.state.borrow().opened_file.clone() {
+                if !self.save_file_with_dialog(&open_file) {
+                    // File was not dealt with (dialog was closed) return early or we will lose unsaved work
+                    // Re-select the file that was open to make it clear the user's action was cancelled
+                    self.gui.account.sidebar.tree.select(&open_file.id);
```

Comment:
The getter for `tree` is used 6 lines later. If we are going to the no getter route, this should probably be taken care of in an immediate followup PR.

Reply:
did this in this PR, in a future pr I can get rid of the other getters: 2f30396